assets/app/features/upgrade/controllers/upgrade-landing.controller.js

```diff
@@ -133,16 +146,28 @@
                     //Success handler. Al precheck passed successfully:
                     function(response) {
 
+                        var checksValidity = [];
+
                         _.forEach(response.data.checks, function(value, key) {
                             // skip unknown checks returned from backend
                             if (key in vm.prechecks.checks) {
                                 vm.prechecks.checks[key].status = value.passed;
+                                if (value.required == true) {
+                                    checksValidity.push(value.passed)
+                                }
                             }
                         });
 
-                        // Update prechecks validity
-                        // TODO: handle disruptive vs non-disruptive properly in the new design
-                        vm.prechecks.valid = (response.data.best_method != 'none');
+                        vm.prechecks.valid = checksValidity.every(function (check) {
```

Comment:
The values stored in `checksValidity` are not "checks" but info on the "passed" status so again if you renamed the argument from `check` to e.g. `checkPassed` the test function would not need an explicit comparison to `true`.